--- judy-1.0.5/src/Judy1/Makefile.in	2015-08-24 01:25:48.300225262 +0800
+++ judy-1.0.5/src/Judy1/Makefile.in	2015-08-24 01:26:01.894138026 +0800
@@ -560,7 +560,7 @@
 
 
 Judy1Tables.c: Judy1TablesGen.c
-	$(CC) $(INCLUDES) $(AM_CFLAGS) @CFLAGS@ -o Judy1TablesGen Judy1TablesGen.c; ./Judy1TablesGen
+	gcc $(INCLUDES) $(AM_CFLAGS) -Os -pipe -o Judy1TablesGen Judy1TablesGen.c; ./Judy1TablesGen
 
 Judy1ByCount.c:../JudyCommon/JudyByCount.c
 	cp -f ../JudyCommon/JudyByCount.c      		Judy1ByCount.c   
--- judy-1.0.5/src/JudyL/Makefile.in	2015-08-24 01:25:16.441116630 +0800
+++ judy-1.0.5/src/JudyL/Makefile.in	2015-08-24 01:25:28.467270312 +0800
@@ -560,7 +560,7 @@
 
 
 JudyLTables.c: JudyLTablesGen.c
-	$(CC) $(INCLUDES) $(AM_CFLAGS) @CFLAGS@ -o JudyLTablesGen JudyLTablesGen.c; ./JudyLTablesGen 
+	gcc $(INCLUDES) $(AM_CFLAGS) -Os -pipe -o JudyLTablesGen JudyLTablesGen.c; ./JudyLTablesGen 
 
 JudyLByCount.c:   ../JudyCommon/JudyByCount.c
 	cp -f ../JudyCommon/JudyByCount.c      		JudyLByCount.c   
