--- a/src/log4c/category.c	2007-09-25 03:12:16.000000000 +0800
+++ b/src/log4c/category.c	2011-08-12 01:33:57.000000000 +0800
@@ -287,3 +287,11 @@
   return a_string;
 }
 
+LOG4C_API void log4c_category_vlog2(const log4c_category_t* a_category,
+        int a_priority,
+        const char* a_format,
+        va_list a_args)
+{
+    const log4c_location_info_t locinfo = LOG4C_LOCATION_INFO_INITIALIZER(NULL);
+    __log4c_category_vlog(a_category, &locinfo, a_priority, a_format, a_args);
+}
--- a//src/log4c-api.h	1970-01-01 08:00:00.000000000 +0800
+++ b//src/log4c-api.h	2011-08-12 01:33:57.000000000 +0800
@@ -0,0 +1,67 @@
+/* $Id$
+ *
+ * log4c_api.h
+ *
+ * Copyright 2001-2002, Meiosys (www.meiosys.com). All rights reserved.
+ *
+ * See the COPYING file for the terms of usage and distribution.
+ */
+
+
+#ifndef _log4c_api_h
+#define _log4c_api_h
+
+#include <stdarg.h>
+#include <log4c/defs.h>
+
+__LOG4C_BEGIN_DECLS
+
+/**
+ * log4c category class
+ **/
+typedef struct __log4c_category	log4c_category_t;
+
+/**
+* Predefined Levels of priorities. These correspond to the priority levels
+* used by syslog(3).
+**/
+typedef enum {
+	/** fatal */	LOG4C_PRIORITY_FATAL	= 000, 
+	/** alert */	LOG4C_PRIORITY_ALERT	= 100, 
+	/** crit */	    LOG4C_PRIORITY_CRIT		= 200, 
+	/** error */	LOG4C_PRIORITY_ERROR	= 300, 
+	/** warn */	    LOG4C_PRIORITY_WARN		= 400, 
+	/** notice */	LOG4C_PRIORITY_NOTICE	= 500, 
+	/** info */	    LOG4C_PRIORITY_INFO		= 600, 
+	/** debug */	LOG4C_PRIORITY_DEBUG	= 700,
+	/** trace */	LOG4C_PRIORITY_TRACE	= 800,
+	/** notset */	LOG4C_PRIORITY_NOTSET	= 900,
+	/** unknown */	LOG4C_PRIORITY_UNKNOWN	= 1000
+} log4c_priority_level_t;
+
+
+LOG4C_API int log4c_init(void);
+
+LOG4C_API int log4c_fini(void);
+
+LOG4C_API void log4c_category_log_name(const char* category_name,
+								  int a_priority,
+								  const char* a_format,
+								  ...);
+
+LOG4C_API void log4c_category_vlog_name(const char* category_name,
+									   int a_priority,
+									   const char* a_format,
+									   va_list a_args);
+
+LOG4C_API log4c_category_t* log4c_category_get(const char* a_name);
+
+LOG4C_API void log4c_category_vlog2(const log4c_category_t* a_category,
+				       int a_priority,
+				       const char* a_format,
+					   va_list a_args);
+__LOG4C_END_DECLS
+
+#endif //_log4c_api_h
+
+
--- a//src/log4c-wrapper.h	1970-01-01 08:00:00.000000000 +0800
+++ b//src/log4c-wrapper.h	2011-08-12 01:33:57.000000000 +0800
@@ -0,0 +1,56 @@
+
+
+#pragma once
+
+//#define WITH_LOG4C
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef WITH_LOG4C
+	#include <log4c-api.h>
+	#include <stdarg.h>
+
+	#ifndef LOG_CATEGORY_NAME_DEFAULT
+		#define LOG_CATEGORY_NAME_DEFAULT "root"
+	#endif
+
+	#ifdef LOG_CATEGORY_NAME
+		#define LOG_CATEGORY_NAME_REAL  LOG_CATEGORY_NAME
+	#else
+		#define LOG_CATEGORY_NAME_REAL  LOG_CATEGORY_NAME_DEFAULT
+	#endif
+
+	#define LOG_PRIORITY_ERROR	LOG4C_PRIORITY_ERROR
+	#define LOG_PRIORITY_WARN	LOG4C_PRIORITY_WARN
+	#define LOG_PRIORITY_NOTICE	LOG4C_PRIORITY_NOTICE
+	#define LOG_PRIORITY_DEBUG	LOG4C_PRIORITY_DEBUG
+	#define LOG_PRIORITY_TRACE	LOG4C_PRIORITY_TRACE
+
+	static void log_proxy(int a_priority, const char* a_format, ...)
+	{
+		static log4c_category_t *category;
+		va_list va;
+		if(category == 0) {
+			category = log4c_category_get(LOG_CATEGORY_NAME_REAL);
+			if(category == 0) return;
+		}
+		va_start(va,a_format);    
+		log4c_category_vlog2(category, a_priority, a_format, va);
+		va_end(va);
+	}
+
+	#define LOGINIT	log4c_init
+	#define LOGFINI	log4c_fini
+	#define LOGMSG	log_proxy
+#else
+	#define LOGINIT()	1
+	#define LOGFINI()	1
+	#define LOGMSG(...)	1
+#endif //WITH_LOG4C
+
+#ifdef __cplusplus
+}; //extern "C"
+#endif
+
--- a//src/Makefile.am	2007-09-25 03:12:16.000000000 +0800
+++ b//src/Makefile.am	2011-08-12 01:33:57.000000000 +0800
@@ -1,4 +1,7 @@
 SUBDIRS = sd log4c
 
 include_HEADERS = \
-	log4c.h
\ No newline at end of file
+	log4c.h \
+	log4c-api.h \
+	log4c-wrapper.h
+
--- a//src/Makefile.in	2007-09-25 03:15:18.000000000 +0800
+++ b//src/Makefile.in	2011-08-12 01:33:57.000000000 +0800
@@ -191,7 +191,9 @@
 top_srcdir = @top_srcdir@
 SUBDIRS = sd log4c
 include_HEADERS = \
-	log4c.h
+	log4c.h \
+	log4c-api.h \
+	log4c-wrapper.h 
 
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
