--- a/evutil_rand.c	2017-02-07 10:46:04.000000000 -0800
+++ b/evutil_rand.c	2017-02-07 11:05:23.000000000 -0800
@@ -40,6 +40,10 @@
 #include "util-internal.h"
 #include "evthread-internal.h"
 
+#ifdef __ANDROID__
+#undef EVENT__HAVE_ARC4RANDOM
+#endif
+
 #ifdef EVENT__HAVE_ARC4RANDOM
 #include <stdlib.h>
 #include <string.h>
@@ -119,7 +123,9 @@
 #ifdef EVENT__ssize_t
 #define ssize_t EVENT__ssize_t
 #endif
+#ifndef __ANDROID__
 #define ARC4RANDOM_EXPORT static
+#endif
 #define ARC4_LOCK_() EVLOCK_LOCK(arc4rand_lock, 0)
 #define ARC4_UNLOCK_() EVLOCK_UNLOCK(arc4rand_lock, 0)
 #ifndef EVENT__DISABLE_THREAD_SUPPORT
